File: WeekendWarriors/Questionnaires/Questionnaires/QLS/AST/Widgets/Text.cs
using Questionnaires.UI.Widgets;
using Questionnaires.QL.AST.Types;

namespace Questionnaires.QLS.AST.Widgets
{
    public class Text : Widget
    {
        public override QuestionWidget CreateWidget(StringType type)
        {
            return new TextBoxWidget();
        }

        public override QuestionWidget CreateWidget(IntegerType type)
        {
            return new IntegerBoxWidget();
        }

        public override QuestionWidget CreateWidget(MoneyType type)
        {
            return new DecimalBoxWidget();
        }
    }
}

File: AlternativeQL/src/org/qls/ast/widget/Text.java
package org.qls.ast.widget;

import org.ql.ast.type.*;

public class Text extends Widget {
    public boolean isCompatibleWith(FloatType type) {
        return true;
    }

    public boolean isCompatibleWith(IntegerType type) {
        return true;
    }

    public boolean isCompatibleWith(MoneyType type) {
        return true;
    }

    public boolean isCompatibleWith(StringType type) {
        return true;
    }

    @Override
    public <T, C> T accept(WidgetVisitor<T, C> visitor, C context) {
        return visitor.visitText(this, context);
    }
}
